express-notes-api/server.ts

```diff
@@ -0,0 +1,154 @@
+import express from 'express';
+import { readFile, writeFile } from 'node:fs/promises';
+
+type Note = {
+  id: number;
+  content: string;
+};
+
+type Data = {
+  nextId: number;
+  notes: Record<number, Note>;
+};
+let data: Data;
+
+type Err = {
+  error: string;
+};
+
+const error: Err = {
+  error: 'An error occured.',
+};
+
+const app = express();
+app.use(express.json());
+
+// Returns list of notes in JSON
+app.get('/api/notes', async (req, res) => {
+  try {
+    await readJSON();
+    const notesList: Note[] = [];
+    for (const key in data.notes) {
+      notesList.push(data.notes[key]);
+    }
+    res.status(200).json(notesList);
+  } catch (err) {
+    console.error(err);
```

Comment:
You need to send an error response to the client. All your catch blocks need to do that.